alerting/alerting-jar/src/main/java/org/hawkular/metrics/alerting/InsertedDataSubscriber.java

```diff
@@ -108,75 +109,75 @@ public void onMetricsServiceReady(@Observes @ServiceReady ServiceReadyEvent even
             }
             events = events.onBackpressureBuffer()
                     .observeOn(Schedulers.io());
-            subscription = events.subscribe(list -> list.forEach(this::onInsertedData));
+            subscription = events.subscribe(list -> onInsertedData(list));
+        }
+    }
+
+    private void onInsertedData(List<Metric<?>> metrics) {
+        List<Data> data = new ArrayList<>(metrics.size());
+        metrics.stream().forEach(m -> addToData(m, data));
+        if (!data.isEmpty()) {
+            try {
+                log.tracef("Sending: %s", data);
```

Comment:
It could be but the difference should not be very much as that test should be performed under the covers via this call.   We're not doing any work preparing the parameter.  Unless there is something I don't understand about how these log 'f' calls work.